### ISSUE TITLE
The sln file generated by cmake should be called "glog.sln" instead of "google-glog.sln"

### ISSUE THREAD

**Issue by jasjuang** [google/glog#149]
I am on windows 10 with visual studio 2015. When I go through the standard procedure of 

``````
git clone https://github.com/google/glog/
mkdir build 
cd build
cmake -A x64 ..
``````

It generates a solution file called google-glog.sln. The problem is when I build the entire solution and build the INSTALL project, glog got installed to C:\Program Files (x86)\google-glog

but

in other google created repository [Ceres](https://github.com/ceres-solver/ceres-solver/blob/master/cmake/FindGlog.cmake) it is expecting glog to be installed in C:\Program Files (x86)\glog, thus causing it to fail. Can someone change the default name of the sln to glog.sln?

**Comment by sergiud** [comment]
You're mixing up two things. The project name and the install location. You want to default the install location on Windows to `%PROGRAMFILES%\glog`. No need to change the project name. You can also change the install location by setting the `CMAKE_INSTALL_PREFIX` CMake variable to the desired directory.

And what exactly fails in your case?

**Comment by jasjuang** [comment]
Yes I understand I can change `CMAKE_INSTALL_PREFIX` in cmake-gui to fix this problem, but the problem is I have to install this in multiple computers and it will be great if this can be automatic.

It fails when I am configuring Ceres with cmake. The FindGlog.cmake in Ceres is expecting glog to be installed in C:\Program Files (x86)\glog instead of  C:\Program Files (x86)\google-glog.

**Comment by sergiud** [comment]
Since you run CMake anyway, you can automate the process by setting `CMAKE_INSTALL_PREFIX` using the command line:
```
cmake <glogdir> -DCMAKE_INSTALL_PREFIX=...
```

**Comment by jasjuang** [comment]
This workaround is acceptable but still slightly inconvenient. Just wondering, what is the main reason that it's not desired to change from google-glog to glog? I filed an issue on Ceres and hopefully they will change if glog is not willing to change. There has to be consistency between Ceres and glog since both are all from Google.

**Comment by sergiud** [comment]
There are no critical reasons besides that this would require to change project's name making it inconsistent with the already provided Visual Studio solution.

However, in general I agree that the install location should default to `%PROGRAMFILES%\glog` which would help `find_package` mechanism to automatically locate glog.

**Comment by jasjuang** [comment]
Thanks. I look forward to the update of changing the default install location to `%PROGRAMFILES%\glog`.

**Comment by jasjuang** [comment]
Ceres was able to [accommodate](https://github.com/ceres-solver/ceres-solver/issues/250) and add in another search option for `%PROGRAMFILES%\google-glog`, but I still feel naming it `%PROGRAMFILES%\glog` is more right though

**Comment by alexsmac** [comment]
Hi, as one of the Ceres' maintainers I thought I would add my 2c here.  In general, CMake best-practice is that the project() name should really match the installed name for precisely this reason, so I think the correct solution is to change project(google-glog) to project(glog).

An interim solution for clients is to use the NAMES option to find_package() as follows:

`
find_package(glog NAMES google-glog glog)
`

this forces find_package() to repeat the search of all prefixes using the names 'google-glog' and 'glog' when expanding each prefix.  Unfortunately both must be specified, as 'google-glog' is required for Windows (by default) and 'glog' is required for all other OSs.

I've pushed a CL for Ceres which does exactly this, once we have confirmation from @jasjuang that it works on Windows (as none of the Ceres maintainers use Windows) we will merge it.

**Comment by sergiud** [comment]
Fixed in #152.
